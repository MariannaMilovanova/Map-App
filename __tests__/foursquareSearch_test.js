import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import FoursquareSearch from '../src/javascripts/components/map/foursquareSearch/foursquareSearch';

it('should search venues based on the value in the state', () => {
    const component = mount(<FoursquareSearch searchVenues={() => {}}/>);
    const preventDefault = jest.fn();
    component.setState({
      term:'bar'
    });
    component.find('form').simulate('submit', { preventDefault });
    expect(toJson(component)).toMatchSnapshot();
    expect(preventDefault).toBeCalled();
});


it('should pass a selected value to the onChange function', () => {
    const component = mount(<FoursquareSearch />);
    component.find('input').simulate('change', { target: {
      value: 'Change function' }
    });

    expect(toJson(component)).toMatchSnapshot();
});
