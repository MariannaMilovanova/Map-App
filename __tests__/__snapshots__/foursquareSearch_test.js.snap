// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass a selected value to the onChange function 1`] = `
<FoursquareSearch>
  <div
    className="fsq-input-search"
  >
    <form
      onSubmit={[Function]}
    >
      <Input
        fluid={true}
        onChange={[Function]}
        placeholder="Search for venues"
        type="text"
        value="Change function"
      >
        <div
          className="ui fluid input"
        >
          <input
            disabled={undefined}
            key="text"
            onChange={[Function]}
            placeholder="Search for venues"
            tabIndex={undefined}
            type="text"
            value="Change function"
          />
        </div>
      </Input>
    </form>
  </div>
</FoursquareSearch>
`;

exports[`should search venues based on the value in the state 1`] = `
<FoursquareSearch
  searchVenues={[Function]}
>
  <div
    className="fsq-input-search"
  >
    <form
      onSubmit={[Function]}
    >
      <Input
        fluid={true}
        onChange={[Function]}
        placeholder="Search for venues"
        type="text"
        value=""
      >
        <div
          className="ui fluid input"
        >
          <input
            disabled={undefined}
            key="text"
            onChange={[Function]}
            placeholder="Search for venues"
            tabIndex={undefined}
            type="text"
            value=""
          />
        </div>
      </Input>
    </form>
  </div>
</FoursquareSearch>
`;
